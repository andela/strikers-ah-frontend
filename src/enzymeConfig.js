import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-localstorage-mock';

export default Enzyme.configure({
  adapter: new Adapter(),
});
