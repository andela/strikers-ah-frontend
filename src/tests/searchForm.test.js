import React from 'react';
import { shallow } from 'enzyme';
import '../enzymeConfig';
import SearchForm from '../components/search/searchForm';

describe('TEST SEARCH FORM', () => {
  let wrapper;
  let instance;
  beforeEach(() => {
    const props = {
      search: jest.fn(),
      history: [],
    };
    wrapper = shallow(<SearchForm {...props} />);
    instance = wrapper.instance();
  });

  test('Should render search Form', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('On value change', () => {
    instance.handleValueChange({
      target: { name: 'filter', value: 'keyword' },
    });
  });

  test('On submit', () => {
    instance.handleSubmit({
      preventDefault: jest.fn(),
    });
  });

  test('When searching on search page', () => {
    instance.setState({ isOnSearchPage: true });
    instance.handleSubmit({
      preventDefault: jest.fn(),
    });
  });

  test('When searching on search page', () => {
    instance.setState({ isOnSearchPage: false });
    const button = wrapper.find('button[type="submit"]');
    button.simulate('click', { preventDefault: jest.fn() });
    instance.handleSubmit({
      preventDefault: jest.fn(),
    });
  });
});
