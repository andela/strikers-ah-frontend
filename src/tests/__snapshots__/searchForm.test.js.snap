// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST SEARCH FORM Should render search Form 1`] = `ShallowWrapper {}`;
